concordium-consensus/src/Concordium/GlobalState/Basic/TreeState.hs

```diff
@@ -200,21 +201,28 @@ instance (bs ~ BlockState m, BS.BlockStateStorage m, Monad m, MonadIO m, MonadSt
                 Nothing -> do
                     possiblyPendingQueue .= ppq
                     return Nothing
+                    
     wipePendingBlocks = do
         possiblyPendingTable .= HM.empty
         possiblyPendingQueue .= MPQ.empty
     getFocusBlock = use focusBlock
     putFocusBlock bb = focusBlock .= bb
     getPendingTransactions = use pendingTransactions
     putPendingTransactions pts = pendingTransactions .= pts
-    getAccountNonFinalized addr nnce =
-            use (transactionTable . ttNonFinalizedTransactions . at' addr) >>= \case
-                Nothing -> return []
-                Just anfts ->
-                    let (_, atnnce, beyond) = Map.splitLookup nnce (anfts ^. anftMap)
-                    in return $ case atnnce of
-                        Nothing -> Map.toAscList beyond
-                        Just s -> (nnce, s) : Map.toAscList beyond
+    getAccountNonFinalized addr nnce = do
+      table <- use transactionTable
+      use (transactionTable . ttNonFinalizedTransactions . at' addr) >>= \case
+        Nothing -> return []
+        Just anfts ->
+          let (_, atnnce, beyond) = Map.splitLookup nnce (anfts ^. anftMap)
+          in return $ case atnnce of
+          Nothing ->
+            let withStatuss = Map.map (Set.map (\tx -> withStatus tx table)) beyond
```

Comment:
I'm really not a fan of this mapping since it involves another lookup in the transaction table. And the whole point of storing the transaction itself in the anftMap is that we don't have to look it up.